src/VecSim/memory/vecsim_malloc.h

```diff
@@ -63,11 +42,77 @@ struct VecSimAllocator {
         return a.allocated != b.allocated;
     }
 
+    static void setMemoryFunctions(VecSimMemoryFunctions memFunctions);
+
 private:
     // Retrive the original requested allocation size. Required for remalloc.
     inline size_t getPointerAllocationSize(void *p) { return *(((size_t *)p) - 1); }
 };
 
+/**
+ * @brief Global function to call for allocating memory buffer (malloc style).
+ *
+ * @param n - Amount of bytes to allocate.
+ * @return void* - Allocated buffer.
+ */
+inline void *vecsim_malloc(size_t n) { return VecSimAllocator::memFunctions.allocFunction(n); }
+
+/**
+ * @brief Global function to call for allocating memory buffer initiliazed to zero (calloc style).
+ *
+ * @param nelem Number of elements.
+ * @param elemsz Element size.
+ * @return void* - Allocated buffer.
+ */
+inline void *vecsim_calloc(size_t nelem, size_t elemsz) {
+    return VecSimAllocator::memFunctions.callocFunction(nelem, elemsz);
+}
+
+/**
+ * @brief Global function to reallocate a buffer (realloc style).
+ *
+ * @param p Allocated buffer.
+ * @param n Number of bytes required to the new buffer.
+ * @return void* Allocated buffer with size >= n.
+ */
+inline void *vecsim_realloc(void *p, size_t n) {
+    return VecSimAllocator::memFunctions.reallocFunction(p, n);
+}
+
+/**
+ * @brief Global function to free an allocated buffer.
+ *
+ * @param p Allocated buffer.
+ */
+inline void vecsim_free(void *p) { VecSimAllocator::memFunctions.freeFunction(p); }
+
+/**
+ * @brief Global function copy a string (strdup style).
+ *
+ * @param s Source string.
+ * @return char* Copied string.
+ */
+inline char *vecsim_strdup(const char *s) {
+    return VecSimAllocator::memFunctions.strdupFunction(s);
+}
```

Comment:
Why did you decide that for `vecsim_strdup` you'll have a `strdupFunction`, but `vecsim_strndup` is using `vecsim_malloc`? Consider making it consistent (I would go for using `vecsim_malloc` in both)